### ISSUE TITLE
No database selected when using em.createNativeQuery()

### ISSUE THREAD

**Issue by uyak** [todvora/jpa-multitenancy#2]
When using nativeQuery throws No database selected error.

**Comment by uyak** [comment]
After some research I believe that this is caused because native query use the driver url directly instead of default schema property, maybe it's necesary to specify the driver url property with schema name included when the EMF are registered

**Comment by todvora** [comment]
Yes, it should be as you say, default database provided already in the connection string. Then it probably depends on the DB type and driver. In the current setting of the example, this works as expected, without any modifications:

``` java
final Query nativeQuery = entityManager.createNativeQuery("SELECT COUNT(*) FROM {h-schema}TODOENTRY");
final Object res = nativeQuery.getSingleResult();
logger.info("Total count is: " + res);
```

**Comment by uyak** [comment]
Yes, but this forces to rewrite all nativeQuerys adding the from clause which was no needed before, and breaks the Tenant unaware EJBs goal, I'm still looking for another solution

**Comment by todvora** [comment]
You are right. I am using this pattern for such a long time now, that I forget, that it could be usable also without  the `{h-schema}` token. I'll try to come up with a solution for this. If you find anything helpful, please let me know (or even better - send a PR). 

Regards,
Tomas

**Comment by uyak** [comment]
A good start poin is implement a custom connection provider:
`<property name="hibernate.connection.provider_class" value="com.tenant.util.MyHibernateConnectionProvider" />`
 and override the getConnection method:
`@Override
    public Connection getConnection() throws SQLException {
        if ( !available ) {
            throw new HibernateException( "Provider is closed!" );
        }
        return useCredentials ? dataSource.getConnection( user, pass ) : dataSource.getConnection();
    }`
but it's not that easy to add a connection to the pool managed by container datasource, because datasource class is hard encapsulated and other jboss classes involved too, finally I used {h-schema} and kept this on things to do one day ;)
Thank you Tomas,
Javi
